Neos.Media/Classes/Domain/Repository/ImageVariantRepository.php

```diff
@@ -0,0 +1,66 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Media\Domain\Repository;
+
+/*
+ * This file is part of the Neos.Media package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Media\Domain\Model\ImageVariant;
+
+/**
+ * A repository for ImageVariants
+ *
+ * @Flow\Scope("singleton")
+ */
+class ImageVariantRepository extends AssetRepository
+{
+    /**
+     * Returns array of ImageVariants with outdated presets
+     *
+     * @param String[] $configuredPresets
```

Comment:
You actually expect a dictionary/map here so this might be misleading, as the key is quite important and it cannot be a simple array. I think it would be good to use the `\Neos\Media\Domain\ValueObject\Configuration\Variant` and VariantPreset value objects to move those Presets around. Maybe add a PresetCollection even.